File: example/env/flutter/material/floatingActionButton.ts
import { Widget } from '../widget';
import { StatelessWidget } from './../widgets/statelessWidget';

interface FloatingActionButtonProps {
    child: Widget;
    onPressed: () => void;
}

declare const flutter: {
    material: {
        floatingActionButton: (this: void, props: FloatingActionButtonProps) => { tag: string };
    }
}

export class FloatingActionButton extends StatelessWidget {
    public props: FloatingActionButtonProps;
    public constructor(props: FloatingActionButtonProps) {
        super();
        this.props = props;
    }

    public build(): Widget {
        return flutter.material.floatingActionButton(this.props);
    }
}

File: lib/builtins/flutter/material/floatingActionButton.dart
import 'package:hydro_sdk/hydroState.dart';
import 'package:hydro_sdk/vm/closure.dart';
import 'package:hydro_sdk/vm/context.dart';
import 'package:hydro_sdk/builtins/flutter/syntheticBox.dart';
import 'package:hydro_sdk/vm/table.dart';
import 'package:flutter/material.dart';

loadFloatingActionButton(
    {@required HydroState luaState, @required HydroTable table}) {
  table["floatingActionButton"] = makeLuaDartFunc(func: (List<dynamic> args) {
    return [
      FloatingActionButton(
        key: maybeUnwrapAndBuildArgument<Widget>(args[0]["key"],
            parentState: luaState),
        child: maybeUnwrapAndBuildArgument<Widget>(args[0]["child"],
            parentState: luaState),
        tooltip: args[0]["tooltip"],
        onPressed: () {
          Closure closure = args[0]["onPressed"];
          closure.dispatch([], parentState: luaState);
        },
      )
    ];
  });
}
